src/server/BaseServer.js

```diff
@@ -116,7 +116,7 @@ export class BaseServer extends EventEmitter {
       let retries = 1;
       if (this._project.kill && await utils.checkPortInUse(this._port, this._addr)) {
         try {
-          const res = await getFetch()(`${this._scheme}://${this._addr}:${this._port}/.kill`);
+          const res = await getFetch(this._project.allowInsecure)(`${this._scheme}://${this._addr}:${this._port}/.kill`);
```

Comment:
this is a request back to localhost, so if https, it most probably is a self-signed, to I would use `true` by default